packages/core/src/modules/connections/DidExchangeProtocol.ts

```diff
@@ -306,39 +321,24 @@ export class DidExchangeProtocol {
       })
     }
 
-    if (!isDid(message.did, 'peer')) {
-      throw new DidExchangeProblemReportError(
-        `Message contains unsupported did ${message.did}. Supported dids are [did:peer]`,
-        {
-          problemCode: DidExchangeProblemReportReason.ResponseNotAccepted,
-        }
-      )
-    }
-    const numAlgo = getNumAlgoFromPeerDid(message.did)
-    if (numAlgo !== PeerDidNumAlgo.GenesisDoc) {
-      throw new DidExchangeProblemReportError(
-        `Unsupported numalgo ${numAlgo}. Supported numalgos are [${PeerDidNumAlgo.GenesisDoc}]`,
-        {
-          problemCode: DidExchangeProblemReportReason.ResponseNotAccepted,
-        }
-      )
-    }
-
-    const didDocument = await this.extractDidDocument(
-      messageContext.agentContext,
+    // Get DID Document either from message (if it is a did:peer) or resolve it externally
+    const didDocument = await this.resolveDidDocument(
+      agentContext,
       message,
       outOfBandRecord
         .getTags()
         .recipientKeyFingerprints.map((fingerprint) => Key.fromFingerprint(fingerprint).publicKeyBase58)
     )
+
     const didRecord = new DidRecord({
-      did: message.did,
+      did: didDocument.id,
       role: DidDocumentRole.Received,
       didDocument,
       tags: {
         // We need to save the recipientKeys, so we can find the associated did
         // of a key when we receive a message from another connection.
         recipientKeyFingerprints: didDocument.recipientKeys.map((key) => key.fingerprint),
+        alsoKnownAs: didDocument.alsoKnownAs,
```

Comment:
I need to think a bit more about that, because it should not be a problem when dealing with peer dids (as it is our agent who will be generating the didDoc based on the spec, and the `alsoKnownAs` is only populated where applicable) but it might be an issue in case we get a DID Exchange Request using a public/custom did.

Reply:
Thanks for popping in here @dbluhm! 

So we need to make sure we've verified the relation of alsoKnownAs both ways before we allow it to be queried.

Maybe we can rename it to verifiedAlsoKnownAs or something different like relatedDids you mentioned ariel.